update(Profile): Improve profile page in setting

hyj1204 opened the pull request:
<!--  Thanks for sending a pull request!-->

### What this PR does 📖
- Add editing username feature
- Improve UI
- Add error messages in username and status message.

**Username:**
1. Length between 4-32 characters.
2. No space in the beginning and end(Trimed). Allow space in the middle of text.
3. @ is not allowed.
4. Only allow alphabetic, digits and punctuation(No emoji).
![image](https://user-images.githubusercontent.com/14248245/203433093-9ea0318c-bb2a-4b13-acac-fcfb0c981ae9.png)
https://docs.rs/regex/latest/regex/
![Screenshot 2022-11-22 at 4 23 36 PM](https://user-images.githubusercontent.com/14248245/203433588-71bcda92-69e3-445f-bcbe-44c6647f93d0.png)

**Status Message:**
1. Length between 0-128 characters.
2. No space in the beginning and end(Trimed). Allow space in the middle of text.


https://user-images.githubusercontent.com/14248245/203436650-af92bedf-5272-4805-bd1e-70f650a02b5d.mov


### Which issue(s) this PR fixes 🔨
- Resolve #307 
<!--Add the ticket Github number such as #Resolve #001 to automatically link the PR to the issue-->

### Special notes for reviewers 🗒️


### Additional comments 🎤

hyj1204 committed:
feat:add basic layout for user profile in setting

hyj1204 committed:
add(Input):add on_enter props in Input component

hyj1204 committed:
clean the code

hyj1204 committed:
add(Profile):add user_name_error and update on_enter function for changing status msg

hyj1204 committed:
chore:uncomment the code during rebasing

hyj1204 committed:
add(Profile):add regex to validate the username input

hyj1204 committed:
refactor(Profile):refactor update_status_msg function

hyj1204 committed:
refactor(Profile):refactor update_user_name function

hyj1204 committed:
chore:update user_name to username

hyj1204 committed:
chore:improve UI

hyj1204 committed:
refactor(Profile): separate components

hyj1204 committed:
refactor(Profile):refactor strings

hyj1204 committed:
fix:add clear error message

hyj1204 committed:
chore: fix fmt

phillsatellite commented:
Super cool! Just one minor little change:
In Status box I think it should say "Type your status message" instead of "Digit" 

<img width="519" alt="Screenshot 2022-11-22 at 6 44 01 PM" src="https://user-images.githubusercontent.com/93608357/203443654-9167d881-2f83-4b64-a71e-69a75a38e5fd.png">

And one small question, should this change the name at the top as well or will we add that in the future? 
<img width="948" alt="Screenshot 2022-11-22 at 6 44 19 PM" src="https://user-images.githubusercontent.com/93608357/203443706-1a05e352-b87f-459c-8a51-a300fda29f85.png">

hyj1204 committed:
chore: text change

hyj1204 commented:
@phillsatellite 
Thanks for the suggestion. I just changed the text of placeholder both in username and status message.
For the username problem on the top, I think it relate to this issue #368 . I added your comment to it.

hyj1204 committed:
fix:allow space in the middle of username

InfamousVague commented:
Looks good after conflicts are resolved @hyj1204

hyj1204 committed:
chore:add missing code during rebase
